File: app/src/main/java/com/example/pondibrian/ukulimaiq/ListAdapter.java
package com.example.pondibrian.ukulimaiq;

import android.content.Context;
import java.util.ArrayList;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

/**
 * Created by Pondi Brian on 4/11/2018.
 */

public class ListAdapter extends ArrayAdapter<String>{
    int vg;
    String[] items_list;
    Context context;
    public ListAdapter (Context context, int vg, int id, String[] items_list){
        super(context, vg, id, items_list);
        this.context =context;
        ArrayList<String> desc;
        this.items_list = items_list;
        this.vg = vg;
    }
// Holds views of Listview to improve its scrolling performance
    static class ViewHolder{
        public TextView txtModel;
        public TextView txtPrice;

    }
    public  View getView(int position,View convertView, ViewGroup parent){
        View rowView = convertView;
        if (rowView==null){
            LayoutInflater inflater =(LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            rowView = inflater.inflate(vg, parent,false);
            ViewHolder holder = new ViewHolder();
            holder.txtModel = rowView.findViewById(R.id.txtmodel);
            holder.txtPrice = rowView.findViewById(R.id.txtprice);
            rowView.setTag(holder);
        }
        String[] items = items_list[position].split("__");
        ViewHolder holder =(ViewHolder)rowView.getTag();
        holder.txtModel.setText(items[0]);
        holder.txtPrice.setText(items[1]);
        return  rowView;
    }
}

File: app/src/main/kotlin/com/pondipb/pondibrian/ukulimaiq/ListAdapter.kt
package com.pondipb.pondibrian.ukulimaiq

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

class ListAdapter(
    context: Context,
    private val vg: Int,
    id: Int,
    private val itemsList: Array<String>
) : ArrayAdapter<String>(context, vg, id, itemsList) {

    // ViewHolder pattern to improve scrolling performance
    private class ViewHolder {
        lateinit var txtModel: TextView
        lateinit var txtPrice: TextView
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var rowView = convertView
        
        if (rowView == null) {
            val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            rowView = inflater.inflate(vg, parent, false)
            
            val holder = ViewHolder()
            holder.txtModel = rowView.findViewById(R.id.txtmodel)
            holder.txtPrice = rowView.findViewById(R.id.txtprice)
            rowView.tag = holder
        }
        
        val items = itemsList[position].split("__")
        val holder = rowView!!.tag as ViewHolder
        
        if (items.isNotEmpty()) {
            holder.txtModel.text = items[0]
            if (items.size > 1) {
                holder.txtPrice.text = items[1]
            }
        }
        
        return rowView
    }
}
